crates/accelerate/src/quantum_circuit/circuit_data.rs

```diff
@@ -350,6 +358,20 @@ impl CircuitData {
         Ok(())
     }
 
+    /// Invokes callable ``func`` with the positional index and operation
+    /// of each instruction.
+    ///
+    /// Args:
+    ///     func (Callable[[int, :class:`~.Operation`], None]):
+    ///         The callable to invoke.
+    #[pyo3(signature = (func))]
+    pub fn enumerate_ops(&self, py: Python<'_>, func: &PyAny) -> PyResult<()> {
```

Comment:
Do you think there's any potential confusion here with the built-in name `enumerate`, which produces an iterator rather than doing the iteration?  I've seen this method called `mapi` in other languages, though I don't love that.

Reply:
There very well may be. This function is actually unlike `map_ops` in that it does not replace the operation with the result of the supplied function. It's more like `foreach_op` but with an index.

How about either `foreach_op_i` or `foreach_op_index`?

Reply:
Oh, then definitely confusion, because I misread it as a `map`-like operation sorry!  Sounds like a good idea to include `foreach_op` in there somewhere: `foreach_op_indexed` sounds fine to me.

Reply:
Done in 02421b6b708e807ff84d6940588d7b298599b501.